optimum/exporters/tasks.py

```diff
@@ -665,38 +668,43 @@ def determine_framework(model: str, framework: str = None) -> str:
             logger.info(f"Local {framework_map[framework]} model found.")
         else:
             try:
-                AutoModel.from_pretrained(model)
+                url = huggingface_hub.hf_hub_url(model_name, WEIGHTS_NAME, subfolder=subfolder)
+                huggingface_hub.get_hf_file_metadata(url)
                 framework = "pt"
             except Exception:
                 pass
 
             if framework is None:
                 try:
-                    TFAutoModel.from_pretrained(model)
+                    url = huggingface_hub.hf_hub_url(model_name, TF2_WEIGHTS_NAME, subfolder=subfolder)
+                    huggingface_hub.get_hf_file_metadata(url)
                     framework = "tf"
                 except Exception:
                     pass
 
-            if framework is None:
-                if is_torch_available():
-                    framework = "pt"
-                elif is_tf_available():
-                    framework = "tf"
-                else:
-                    raise EnvironmentError("Neither PyTorch nor TensorFlow found in environment. Cannot export model.")
+        if is_torch_available():
+            framework = framework or "pt"
+        elif is_tf_available():
+            framework = framework or "tf"
+        else:
+            raise EnvironmentError("Neither PyTorch nor TensorFlow found in environment. Cannot export model.")
 
         logger.info(f"Framework not specified. Using {framework} to export to ONNX.")
 
         return framework
 
     @staticmethod
-    def infer_task_from_model(model_name_or_path: str) -> str:
+    def infer_task_from_model(model_name_or_path: str, subfolder: str = "", revision: Optional[str] = None) -> str:
         """
         Infers the task from the model repo.
 
         Args:
             model_name_or_path (`str`):
                 The model repo or local path (not supported for now).
+            subfolder (`str`, *optional*, defaults to `""`):
+                The subfolder where the model is stored.
```

Comment:
```suggestion
                In case the relevant files are located inside a subfolder of the model repo on huggingface.co, you can
                specify the folder name here.
```